### PR TITLE
fix: Resolve pkg lint errors

### PR DESCRIPTION
**Pull request by ahmed-nour-fdc**

### CODE REVIEW

File: pkg/auth/azure.go [hunk 1]
```diff
@@ -110,7 +111,7 @@ func HttpAuthMiddleWare(resp http.ResponseWriter, req *http.Request, jwks *keyfu
 	}
 	if err != nil {
 		resp.WriteHeader(http.StatusUnauthorized)
-		resp.Write([]byte("Invalid authorization header provided"))
+		resp.Write([]byte("Invalid authorization header provided")) //nolint: errcheck
```
Comment (hannesg) [inline pkg/auth/azure.go:114]: You can  make the lint pass by writing it like this:

```
_, _ = resp.Write(...)
```

File: pkg/setup/setup.go [hunk 2]
```diff
@@ -114,7 +113,7 @@ func Run(ctx context.Context, config ServerConfig) {
 	pv, handler, _ := metrics.Init()
 	ctx = metrics.WithProvider(ctx, pv)
 
-	pv.Meter("setup").Int64ObservableGauge("background_job_ready", metric.WithInt64Callback(func(_ context.Context, o metric.Int64Observer) error {
```
Comment (hannesg) [inline pkg/setup/setup.go:117]: What is the lint actually complaining about?

File: pkg/setup/setup.go [hunk 3]
```diff
@@ -167,10 +166,10 @@ func (s *setup) listenToShutdownSignal(ctx context.Context, cancelFunc context.C
...
 
 	// call shutdown hooks
-	gracefulShutdown(ctx, s, 30*time.Second)
+	gracefulShutdown(s, 30*time.Second)
 }
 
-func gracefulShutdown(ctx context.Context, s *setup, timeout time.Duration) {
+func gracefulShutdown(s *setup, timeout time.Duration) {
 	// Instantiate background context
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```
Comment (hannesg) [inline pkg/setup/setup.go:174]: Uh, that is a good catch. I would think that the ctx paramer actually makes sense and we should not use `context.Background()` but `ctx` as first pararmeter here to pass the logger and tracer from the context.

Comment (ahmed-nour-fdc) [reply pkg/setup/setup.go:174]: What I understood from tracing the code is that we don't actually want to use ctx argument as it is probably already cancelled previously, but I might be wrong here.

File: pkg/setup/setup.go (hunk 2 above)
Comment (ahmed-nour-fdc) [reply pkg/setup/setup.go:117]: Return value not checked. I couldn't figure out exactly how to do so.

File: pkg/setup/setup.go (hunk 3 above)
Comment (sven-urbanski-freiheit-com) [reply pkg/setup/setup.go:174]: I haven't tested it, but I would assume that still use a context for logging, even when it's cancelled. In which case we should do what hannes suggested.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**